core-bundle/tests/Image/PictureFactoryTest.php

```diff
@@ -567,6 +567,100 @@ function (?ResizeConfiguration $size): bool {
+    {
+        $path = $this->getTempDir().'/images/dummy.jpg';
+        $imageMock = $this->createMock(ImageInterface::class);
+
+        $pictureGenerator = $this->createMock(PictureGeneratorInterface::class);
+        $pictureGenerator
+            ->method('generate')
+            ->willReturnCallback(
+                function (ImageInterface $image, PictureConfiguration $config, ResizeOptions $options) use ($imageMock, $expected) {
+                    $this->assertSame($expected, $options->getSkipIfDimensionsMatch());
+
+                    return new Picture(['src' => $imageMock, 'srcset' => []], []);
+                }
+            )
+        ;
+
+        $imageFactory = $this->createMock(ImageFactoryInterface::class);
+        $imageFactory
+            ->method('create')
+            ->willReturn($imageMock)
+        ;
+
+        $pictureFactory = $this->getPictureFactory($pictureGenerator, $imageFactory);
+        $pictureFactory->setPredefinedSizes([
+            'size_skip' => [
+                'resizeMode' => ResizeConfiguration::MODE_BOX,
+                'skipIfDimensionsMatch' => true,
+                'items' => [],
+            ],
+            'size_noskip' => [
+                'resizeMode' => ResizeConfiguration::MODE_BOX,
+                'skipIfDimensionsMatch' => false,
+                'items' => [],
+            ],
+        ]);
+
+        $pictureFactory->create($path, $size, $resizeOptions);
+    }
+
+    public function getResizeOptionsScenarios(): \Generator
+    {
+        yield 'prefer skipIfDimensionsMatch from explicitly set options (1)' => [
+            (new ResizeOptions())->setSkipIfDimensionsMatch(true),
+            'size_skip',
+            true,
+        ];
+
+        yield 'prefer skipIfDimensionsMatch from explicitly set options (2)' => [
+            (new ResizeOptions())->setSkipIfDimensionsMatch(true),
+            'size_noskip',
+            true,
+        ];
+
+        yield 'prefer skipIfDimensionsMatch from explicitly set options (3)' => [
+            (new ResizeOptions())->setSkipIfDimensionsMatch(false),
+            'size_skip',
+            false,
+        ];
+
+        yield 'prefer skipIfDimensionsMatch from explicitly set options (4)' => [
+            (new ResizeOptions())->setSkipIfDimensionsMatch(false),
+            'size_noskip',
+            false,
+        ];
+
+        yield 'use skipIfDimensionsMatch from predefined size (1)' => [
+            null,
+            'size_skip',
+            true,
+        ];
+
+        yield 'use skipIfDimensionsMatch from predefined size (2)' => [
+            null,
+            'size_noskip',
+            false,
+        ];
+
+        yield 'fallback to default resize option when passing a picture configuration' => [
+            null,
+            new PictureConfiguration(),
+            false,
+        ];
+
+        yield 'fallback to default predefined size' => [
+            null,
+            null,
+            true, // !
```

Comment:
This comment is not really helpful. Either remove it or explain why using `true` is special here.

Reply:
Agreed. I think I'd remove it, then.

/cc @ausi

Reply:
Removed in e6b564c36645d73fb0ac032a2eabd096387dd67f.

Reply:
Did you mention me to check if `true` is the correct result here?
I think this is correct, especially for BC reasons because generating a picture without any configuration resulted in the original path (not *assets/images*) in previous verions.

Reply:
Yes, just to double check! Thanks!